node/node.go

```diff
@@ -801,7 +799,33 @@ func (n *Node) superviseManager(ctx context.Context, securityConfig *ca.Security
 		timer := time.NewTimer(16 * time.Second)
 		select {
 		case <-timer.C:
-			log.G(ctx).Warn("failed to get worker role after manager stop, restarting manager")
+			if wasRemoved {
+				// We need to be extra careful about restarting
+				// the manager. It may cause the node to
+				// wrongly join under a new Raft ID.
+				log.G(ctx).Warn("failed to get worker role after manager stop, forcing certificate renewal")
+				timer.Reset(16 * time.Second)
+				select {
+				case forceCertRenewal <- struct{}{}:
+					select {
+					case <-timer.C:
+						log.G(ctx).Warn("failed to get worker role after manager stop, restarting manager")
+					case <-workerRole:
+					case <-ctx.Done():
+						timer.Stop()
+						waitRoleCancel()
+						return ctx.Err()
+					}
+				case <-timer.C:
+					log.G(ctx).Warn("failed to trigger certificate renewal after manager stop, restarting manager")
+				case <-ctx.Done():
+					timer.Stop()
```

Comment:
Non-blocking:  there seem to be a lot of `waitRoleCancel()` and `timer.Stop()` calls - not sure if this would be cleaner, but if this code inside the for loop were wrapped in its own function, maybe these could be deferred instead?